src/BugsnagUnity/Delivery.cs

```diff
@@ -524,21 +524,54 @@ private IEnumerator ProcessCachedItems(Type t)
 
                     try
                     {
+                        // if something goes wrong at this stage then we silently discard the file as it's most likley an io error at some point
                         payloadDictionary = ((JsonObject)SimpleJson.DeserializeObject(payloadJson)).GetDictionary();
                     }
                     catch (Exception e)
                     {
-                        Debug.LogException(new Exception("Bugsnag Error. Deserialising a cached payload for delivery failed: " + e.Message + " Cached json: " + payloadJson));
+                        _cacheManager.RemoveCachedEvent(id);
                         continue;
                     }
 
+
+                    // if something goes wrong at this stage then we discard the file and report the error as it might be a bug in the sdk
                     if (isSession)
                     {
-                        Deliver(new SessionReport(_configuration, payloadDictionary));
+                        try
+                        {
+                            var sessionReport = new SessionReport(_configuration, payloadDictionary);
+                            Deliver(sessionReport);
```

Comment:
It feels like the `Deliver` call should be outside the catch as it's conceptually not the same thing as we're trying to report.